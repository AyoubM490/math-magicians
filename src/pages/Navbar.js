import React from 'react';
import { NavLink } from 'react-router-dom';
import Header from '../components/Header';

const Navbar = () => {
  const links = [
    {
      id: 1,
      path: '/',
      text: 'Home',
    },
    {
      id: 2,
      path: '/calculator',
      text: 'Calculator',
    },
    {
      id: 3,
      path: '/quote',
      text: 'Quote',
    },
  ];

  return (
    <nav className="navBar">
      <div className="navBarWrap">
        <Header />
        <ul className="navLinks">
          {links.map((link) => (
            <li key={link.id} className="navLink">
              <NavLink to={link.path} activeClassName="active-link" exact className="link">
                {link.text}
              </NavLink>
            </li>
          ))}
        </ul>
      </div>
    </nav>
  );
};

export default Navbar;
