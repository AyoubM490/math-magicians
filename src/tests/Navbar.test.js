import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import { render, fireEvent, screen } from '@testing-library/react';
import Navbar from '../pages/Navbar';

describe('test if Navbar', () => {
  test('renders correctly', () => {
    const component = renderer
      .create(
        <BrowserRouter>
          <Navbar />
        </BrowserRouter>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});

describe('Navbar', () => {
  test('Responding on Home click ', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Home'));
  });
  test('Responding on Calculator click ', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Calculator'));
  });
  test('Responding on Quote click ', () => {
    render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    fireEvent.click(screen.getByText('Quote'));
  });
});
