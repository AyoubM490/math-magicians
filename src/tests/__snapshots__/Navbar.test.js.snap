// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test if Navbar renders correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="navBarWrap"
  >
    <div>
      <h1
        className="bigHeading"
      >
        Math Magicians
      </h1>
    </div>
    <ul
      className="navLinks"
    >
      <li
        className="navLink"
      >
        <a
          aria-current="page"
          className="link active-link"
          href="/"
          onClick={[Function]}
          style={Object {}}
        >
          Home
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          aria-current={null}
          className="link"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </li>
      <li
        className="navLink"
      >
        <a
          aria-current={null}
          className="link"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
